code/modules/research/xenobiology/crossbreeding/_status_effects.dm

```diff
@@ -740,36 +740,43 @@
 	var/mob/living/clone
 
 /datum/status_effect/stabilized/cerulean/on_apply()
-	var/typepath = owner.type
-	clone = new typepath(owner.loc)
-	var/mob/living/carbon/O = owner
-	var/mob/living/carbon/C = clone
-	if(istype(C) && istype(O))
-		C.real_name = O.real_name
-		C.update_name_tag(C.real_name) // monkestation edit: name tags
-		O.dna.transfer_identity(C)
-		C.updateappearance(mutcolor_update=1)
+	create_clone()
+	RegisterSignal(owner, COMSIG_LIVING_DEATH, PROC_REF(on_owner_death))
 	return ..()
 
-/datum/status_effect/stabilized/cerulean/tick()
-	if(owner.stat == DEAD)
-		if(clone && clone.stat != DEAD)
-			owner.visible_message(span_warning("[owner] blazes with brilliant light, [linked_extract] whisking [owner.p_their()] soul away."),
-				span_notice("You feel a warm glow from [linked_extract], and you open your eyes... elsewhere."))
-			if(owner.mind)
-				owner.mind.transfer_to(clone)
-			clone = null
-			qdel(linked_extract)
-		if(!clone || clone.stat == DEAD)
-			to_chat(owner, span_notice("[linked_extract] desperately tries to move your soul to a living body, but can't find one!"))
-			qdel(linked_extract)
-	..()
-
 /datum/status_effect/stabilized/cerulean/on_remove()
-	if(clone)
+	UnregisterSignal(owner, COMSIG_LIVING_DEATH)
+	if(!QDELETED(clone))
 		clone.visible_message(span_warning("[clone] dissolves into a puddle of goo!"))
 		clone.unequip_everything()
 		qdel(clone)
+	clone = null
+	return ..()
+
+/datum/status_effect/stabilized/cerulean/proc/create_clone()
+	var/typepath = owner.type
+	clone = new typepath(owner.loc)
+	clone.name = clone.real_name = owner.real_name
+	clone.update_name_tag(clone.real_name)
+	var/mob/living/carbon/carbon_owner = owner
+	var/mob/living/carbon/carbon_clone = clone
+	if(istype(carbon_owner) && istype(carbon_clone))
+		carbon_owner.dna.transfer_identity(carbon_clone)
+		carbon_clone.updateappearance(mutcolor_update = TRUE)
```

Comment:
So, this whole code block has been bothering me for the whole day.
If it's a carbon, it should just `dna.transfer_identity`, update appearance and then return.
The only thing missing in DNA would be the name tag, but that seems to be something relevant enough that it could be added to DNA IMO

Also the `clone.name = clone.real_name = owner.real_name` should not be on a single line since that looks confusing to anyone else looking at it.